lib/field_parser.rb

```diff
@@ -14,23 +20,26 @@ def generate_string_representation
         enumeration = _generate_enumeration
         chronology = _generate_chronology
         @string_rep += enumeration.length > 0 ? enumeration : ''
-        if chronology.length > 0
-            chronology = enumeration.length > 0 ? " (#{chronology})" : chronology
-            @string_rep += chronology
-        end
+
+        return unless chronology.length > 0
+
+        chronology = enumeration.length > 0 ? " (#{chronology})" : chronology
+        @string_rep += chronology
     end
 
     private
 
     def _generate_enumeration
-        @@enumeration_codes.split('').map { |c| @h_field.include?(c) ? "#{@y_field[c]} #{@h_field[c]}" : nil }.compact.join(', ')
+        @@enumeration_codes.split('').map do |c|
+            @h_field.include?(c) && _empty_field_check(@h_field[c]) ? "#{@y_field[c]} #{@h_field[c]}" : nil
+        end.compact.join(', ')
     end
 
     def _generate_chronology
         date_component = DateComponent.new
-        components = @@chronology_codes.split('').map do |c|
-            if @h_field.include? c 
-                date_component.set_field(@y_field[c].tr('()', ''), @h_field[c])
+        @@chronology_codes.split('').map do |c|
+            if @h_field.include?(c) && _empty_field_check(@h_field[c])
+                date_component.set_field(_standardize_date_definition_field(@y_field[c]), @h_field[c])
```

Comment:
These two lines are the core of these updates, they ensure that a value is present and that a proper field name can be found. A similar approach is taken in the `_generate_enumeration` method above.

These are the only two real functionality changes in this commit.